tensorboard/plugins/profile/profile_plugin_loader.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Wrapper around plugin to conditionally enable it."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorboard.plugins import base_plugin
+
+
+class ProfilePluginLoader(base_plugin.TBLoader):
+  """HParamsPlugin factory.
+
+  This class checks for `tensorflow` install and dependency.
+  """
+
+  def define_flags(self, parser):
+    group = parser.add_argument_group('profile plugin')
+    group.add_argument(
+        '--master_tpu_unsecure_channel',
+        metavar='ADDR',
+        type=str,
+        default='',
+        help='''\
+IP address of "master tpu", used for getting streaming trace data
+through tpu profiler analysis grpc. The grpc channel is not secured.\
+''')
+
+  def load(self, context):
+    """Returns the plugin, if possible.
+
+    Args:
+      context: The TBContext flags.
+
+    Returns:
+      A ProfilePlugin instance or None if it couldn't be loaded.
+    """
+    try:
+      # pylint: disable=g-import-not-at-top,unused-import
+      import tensorflow
+      # pylint: disable=g-import-not-at-top
+      from tensorboard.plugins.profile.profile_plugin import ProfilePlugin
```

Comment:
This import should stay outside the try-except, I think?  Since if it's not importable that should be a real error.

Reply:
You are right - there was some earlier issue that I've since fixed here.

Reply:
Actually, I still get an error when running locally:

```
(tfpy36) orionr-mbp:tensorboard orionr$ ./bazel-bin/tensorboard/tensorboard --logdir ~/local/sample_runs/runs
Traceback (most recent call last):
  File "/Users/orionr/local/tensorboard/./bazel-bin/tensorboard/tensorboard.runfiles/org_tensorflow_tensorboard/tensorboard/main.py", line 82, in <module>
    run_main()
  File "/Users/orionr/local/tensorboard/./bazel-bin/tensorboard/tensorboard.runfiles/org_tensorflow_tensorboard/tensorboard/main.py", line 78, in run_main
    sys.exit(tensorboard.main())
  File "/Users/orionr/local/tensorboard/bazel-bin/tensorboard/tensorboard.runfiles/org_tensorflow_tensorboard/tensorboard/program.py", line 224, in main
    server = self._make_server()
  File "/Users/orionr/local/tensorboard/bazel-bin/tensorboard/tensorboard.runfiles/org_tensorflow_tensorboard/tensorboard/program.py", line 305, in _make_server
    self.assets_zip_provider)
  File "/Users/orionr/local/tensorboard/bazel-bin/tensorboard/tensorboard.runfiles/org_tensorflow_tensorboard/tensorboard/backend/application.py", line 154, in standard_tensorboard_wsgi
    plugin = loader.load(context)
  File "/Users/orionr/local/tensorboard/bazel-bin/tensorboard/tensorboard.runfiles/org_tensorflow_tensorboard/tensorboard/plugins/profile/profile_plugin_loader.py", line 57, in load
    from tensorboard.plugins.profile.profile_plugin import ProfilePlugin
  File "/Users/orionr/local/tensorboard/bazel-bin/tensorboard/tensorboard.runfiles/org_tensorflow_tensorboard/tensorboard/plugins/profile/profile_plugin.py", line 30, in <module>
    from tensorflow.python.eager import profiler_client
ImportError: cannot import name 'profiler_client'
```

I'll push changes as-is, but this was why I put it inside the try originally.

Reply:
Hmm, what TF version are you trying this with locally?  That code was added recently so it's possible it's only in fairly recent TF nightly releases.

Reply:
I was on 1.13.0, but I just upgraded to 1.14.1 and now get a beholder error:

```
(tfpy36) orionr-mbp:tensorboard orionr$ ./bazel-bin/tensorboard/tensorboard --logdir ~/local/sample_runs/runs
Limited tf.compat.v2.summary API due to missing TensorBoard installation
Traceback (most recent call last):
  File "/Users/orionr/local/tensorboard/./bazel-bin/tensorboard/tensorboard.runfiles/org_tensorflow_tensorboard/tensorboard/main.py", line 41, in <module>
    from tensorboard import default
  File "/Users/orionr/local/tensorboard/bazel-bin/tensorboard/tensorboard.runfiles/org_tensorflow_tensorboard/tensorboard/default.py", line 37, in <module>
    from tensorboard.plugins.beholder import beholder_plugin_loader
  File "/Users/orionr/local/tensorboard/bazel-bin/tensorboard/tensorboard.runfiles/org_tensorflow_tensorboard/tensorboard/plugins/beholder/__init__.py", line 22, in <module>
    from tensorboard.plugins.beholder.beholder import Beholder
  File "/Users/orionr/local/tensorboard/bazel-bin/tensorboard/tensorboard.runfiles/org_tensorflow_tensorboard/tensorboard/plugins/beholder/beholder.py", line 199, in <module>
    class BeholderHook(tf.estimator.SessionRunHook):
AttributeError: module 'tensorflow' has no attribute 'estimator'
```
Should I try 2.0?

Reply:
And, for 2.0, I'm guessing the right install is `pip install tf-nightly-2.0-preview`?

Reply:
Do you mean 1.12.0 and 1.13.1?  There is no TF 1.14.x release yet that I'm aware of.

It should work to just `pip install tf-nightly` in a fresh virtualenv for local testing - this is the same as what Travis does and that's passing, and it just worked for me locally.

Reply:
Confirmed that `tf-nightly-2.0-preview` seems to work

Reply:
This is what I see when doing a fresh `pip install tf-nightly`

```
(tfpy36) orionr-mbp:tensorboard orionr$ pip freeze | grep tf-
tf-estimator-nightly==1.13.0.dev2019012901
tf-nightly==1.14.1.dev20190319
```

Reply:
Okay, I think this is because your tf-estimator-nightly installation is old.  Did you start from a completely fresh virtualenv?  I think if you start completely fresh you should pick up tf-estimator-nightly 1.14.0.dev2019031901 (that's what I get trying it locally).

Reply:
You are correct - `pip install --upgrade tf-estimator-nightly` did the trick. I think we're good then. Thanks!